apiserver/facades/client/application/application.go

```diff
@@ -1746,6 +1700,15 @@ func (api *APIBase) ApplicationsInfo(ctx context.Context, in params.Entities) (p
 			continue
 		}
 
+		locator, err := api.applicationService.GetCharmLocatorByApplicationName(ctx, tag.Name)
```

Comment:
Goodness me; this whole API call is a multitude of service access that should be poked into fewer, ideally one, call.